provider/resource.go

```diff
@@ -119,6 +126,24 @@ func NewResource(id, rt string, p Provider) Resource {
 	}
 }
 
+func (r *resource) AttributesReference() ([]string, error) {
+	var (
+		res = new(tfdocs.Resource)
+		err error
+	)
+	switch r.provider.String() {
+	case "aws":
+		res, err = awsdocs.GetResource(r.resourceType)
+	case "google":
+		res, err = googledocs.GetResource(r.resourceType)
+	}
```

Comment:
This could also be done by having an static map where the key is the `r.provider.String()` and the value `func (r string) (*tfdocresource.Resource, error)`, this way is an static map, and easy to add more instead of having to add a new `case` to the switch.

Reply:
@xescugc good catch ! it's way cleaner. Ready for final review :wink: